### ISSUE TITLE
images in val_img unecessary for test.py?

### ISSUE THREAD

**Issue by jmoberreuter** [NVlabs/SPADE#49]
Hi, great job on this paper! One question: if I am not retraining, just trying to generate images from a sketch, the validation images should not be necessary. However, if I delete the images, there is an error. On the other hand, if I change the images for something completely different, the output result doesn't change. Am I missing something or could the requirement to have validation images be dropped? Thanks a bunch!

**Comment by taesungp** [comment]
For this, you will need to modify the code of the dataset class ([CocoDataset for example](https://github.com/NVlabs/SPADE/blob/master/data/coco_dataset.py)) to be able to load the layouts only. 

The image generation process entirely depends on the label layout, and does not depend on the validation image. Therefore, the output result will not change even if you change the image. The groud-truth RGB image is merely used for comparison.

**Comment by MrWwei** [comment]
@jmoberreuter have you solved the issue？

**Comment by nitish11** [comment]
@jmoberreuter @MrWwei Have you resolved the issue?
@taesungp Could you please explain a bit more about changes to be made.

**Comment by nitish11** [comment]
I got the solution, the dataloader code is using the val_img path for reference. 
[data_i in dataloader](https://github.com/NVlabs/SPADE/blob/master/test.py#L32) is a dictionary with following keys: dict_keys(['label', 'instance', 'image', 'path']) 
data_i[path] contains list of  [name of validation images](https://github.com/NVlabs/SPADE/tree/master/datasets/coco_stuff/val_img), but it is not being used in [Evaluation code](https://github.com/NVlabs/SPADE/blob/master/models/pix2pix_model.py) in inference mode.

**Comment by nitish11** [comment]
I modified the test code and it's generating and saving the images. 
```
# test
opt.how_many = 1
for i, data_i in enumerate(dataloader):
    if i * opt.batchSize >= opt.how_many:
        break
    print('-- Data_i keys and values :',data_i.keys(),type(data_i))
    print([(k, type(data_i[k]), np.shape(data_i[k])) for k in data_i])
    print("-- Path :",data_i['path'])

    data_i['path'] = [None]
    generated = model(data_i, mode='inference')

    generated_image_path = "generated.png"

    for b in range(generated.shape[0]):
        generated_image = generated[b]
        
        # generated_image = np.moveaxis(generated_image.cpu().detach().numpy(), 0, -1)
        generated_image = util.tensor2im(generated_image)
        print('---- generated image ', generated_image_path, np.shape(generated_image))
        cv2.imwrite(generated_image_path, generated_image)
```
